### PR TITLE
EmailValidator: Switching to text template

### PR DESCRIPTION
**Pull request by mintsoft**
###### Description of changes

Ref: https://github.com/duckduckgo/zeroclickinfo-goodies/issues/2767

###### People to notify (@mention interested parties)

@lamanh 

---

Instant Answer Page: https://duck.co/ia/view/email_validator

[Maintainer](http://docs.duckduckhack.com/maintaining/guidelines.html): @stelim

### CODE REVIEW

**Comment by daxtheduck** [comment]
Metadata placeholder post
            
*If you haven't already, please edit your first comment to include a link to the IA page at the bottom.*
            
---
            
**How to find your IA page**
- New Instant Answer
  -- follow the steps [here](https://duck.co/ia/new_ia) to create your IA page.
              
- Existing Instant Answer
  -- find your IA page from this [list](https://duck.co/ia)

File: t/EmailValidator.t [hunk 1]
```diff
@@ -9,65 +9,31 @@ use DDG::Test::Goodie;
...
-    ),
-    'validate my email foo.bar@example.com' => test_zci(
-        re(qr/appears to be valid/),
-        structured_answer => {
-            input     => ['foo.bar@example.com'],
-            operation => 'Email address validation',
-            result    => re(qr/appears to be valid/)
-        }
-    ),
+    'validate my email foo@example.com' => build_test($valid_re, 'Email address validation: foo@example.com'),
```
Comment (mintsoft) [inline t/EmailValidator.t]: @GuiltyDolphin Before you suggest moving the `Email address validation: ` into the `build_test`; I tried that, but the `ignore()` below shafts us

File: t/EmailValidator.t [hunk 2]
```diff
@@ -9,65 +9,31 @@ use DDG::Test::Goodie;
...
-    ),
-    'validate foo@example.lmnop' => test_zci(
-        re(qr/Please check the top-level domain/),
-        structured_answer => {
-            input     => ['foo@example.lmnop'],
-            operation => 'Email address validation',
-            result    => re(qr/Please check the top-level domain/),
-        }
-    ),
+      => build_test(qr/Please check the address/, ignore()),
```
Comment (GuiltyDolphin) [inline t/EmailValidator.t]: @mintsoft Why are we using `ignore` here? We know what the text should be, don't we? (`user@exampleaaaaaaa...`)?

Comment (mintsoft) [inline t/EmailValidator.t]: Just Try it :+1:

Comment (GuiltyDolphin) [inline t/EmailValidator.t]: @mintsoft That was automated :wink:

Looks like an issue with the package; we can still pass in just the domain though (just check the ref, or even have a flag to say 'this errors!').

Comment (mintsoft) [inline t/EmailValidator.t]: @GuiltyDolphin How do you suggest to do that? Every way I've reshaped this ends up with some variety of that error

Comment (GuiltyDolphin) [inline t/EmailValidator.t]: @mintsoft You could have something like:

```perl
my ($re, $address, $ignore) = @_;

$adress = $ignore ? ignore() : $address;
...
```

In the builder, then pass `build_test(qr/.../, '', 1)` for the test that should have the subtitle ignored?

Comment (mintsoft) [inline t/EmailValidator.t]: Ah yeah; I'll do that

**Comment by GuiltyDolphin** [comment]
@mintsoft Thanks, LGTM :+1:

Might be worth raising an issue about the error?

**Pull request merged**